src/common/src/types/num256.rs

```diff
@@ -227,6 +260,36 @@ impl Int256 {
     }
 }
 
+impl UInt256Ref<'_> {
+    pub fn memcmp_serialize(
+        &self,
+        serializer: &mut memcomparable::Serializer<impl bytes::BufMut>,
+    ) -> memcomparable::Result<()> {
+        // Big-endian rendering of the full 256-bit value.
+        // We push the 32 bytes one-by-one with `serialize_u8`, which the
+        // docs guarantee writes a *single* raw byte – no length tags,
+        // no terminators. :contentReference[oaicite:0]{index=0}
+        for byte in self.0.to_be_bytes() {
```

Comment:
For Int256, it is serialized by 
```
let (hi, lo) = self.0.into_words();
(hi, lo as u128).serialize(serializer)
```
This is the same as how int256 is compared with each other in the ethum crate:
https://github.com/nlordell/ethnum-rs/blob/main/src/int/cmp.rs#L34-L36
```
let (ahi, alo) = self.into_words();
let (bhi, blo) = other.into_words();
(ahi, alo as u128).cmp(&(bhi, blo as u128))
```

curious why uint256 is not done in the same way by following how it is compared with each other in the ethum crate:
https://github.com/nlordell/ethnum-rs/blob/main/src/uint/cmp.rs#L33
```
self.into_words().cmp(&other.into_words())
```

Reply:
Thanks!